tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,946 @@
+#!/usr/bin/python
+""" The Python Hipify script.
+##
+# Copyright (c) 2015-2016 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+
+import argparse
+import constants
+import re
+import shutil
+import sys
+import os
+import yaml
+
+from functools import reduce
+from cuda_to_hip_mappings import CUDA_TO_HIP_MAPPINGS
+
+def openf(filename, mode):
+   if sys.version_info[0] == 3:
+      return open(filename, mode, errors='ignore')
+   else:
+      return open(filename, mode)
+
+# Color coding for printing
+class bcolors:
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+def update_progress_bar(total, progress):
+    """
+    Displays and updates a console progress bar.
+    """
+    barLength, status = 20, ""
+    progress = float(progress) / float(total)
+    if progress >= 1.:
+        progress, status = 1, "\r\n"
+
+    # Number of blocks to display. Used to visualize progress.
+    block = int(round(barLength * progress))
+    text = "\r[{}] {:.0f}% {}".format(
+        "#" * block + "-" * (barLength - block), round(progress * 100, 0),
+        status)
+
+    # Send the progress to stdout.
+    sys.stdout.write(text)
+
+    # Send the buffered text to stdout!
+    sys.stdout.flush()
+
+
+def filename_ends_with_extension(filename, extensions):
+    """Helper method to see if filename ends with certain extension"""
+    return reduce(lambda result, ext: filename.endswith("." + ext) or result, extensions, False)
```

Comment:
Oh man, Guido would totally be rolling in his... bed... at this code. ;)

Reply:
leave or fix this now?

Reply:
fine for now

Reply:
Do you think we could get code reviews from Guido?